frontend/source/js/data-capture/step-bar.js

```
@@ -1,32 +1,83 @@
-/* global jQuery, document */
+/* global window, document */
+import 'document-register-element';
 
-const $ = jQuery;
+const STEP_CLASS = '.step-bar__bubbles li';
+const STEP_LABEL_CLASS = '.step-bar__labels li';
+const CURRENT_STEP_CLASS = '.step-bar__labels .current';
 
-$(document).ready(() => {
-  const steps = document.querySelectorAll('.step-bar__bubbles li');
-  const stepLabels = document.querySelectorAll('.step-bar__labels li');
-  const currentStep = document.querySelector('.step-bar__labels .current');
+class StepBar extends window.HTMLElement {
+  attachedCallback() {
+    const steps = this.querySelectorAll(STEP_CLASS);
+    const stepLabels = this.querySelectorAll(STEP_LABEL_CLASS);
+    const currentStep = this.querySelector(CURRENT_STEP_CLASS);
 
-  const hidePrevLabel = () => {
-    const prevLabel = document.querySelector('.step-bar__labels .js-show');
+    if (steps && stepLabels && currentStep) {
```

It's great that you're checking to make sure all these DOM bits are truthy here.

Part of me wonders if we should log an error or something if these things don't all exist, rather than failing silently, but I'm not sure what we do in other places in our code (it's been too long since I touched the web componenty bits of this code lol).   Ah well, we can just deal with it later, if at all.